index.js

```diff
@@ -48,11 +50,47 @@ function spawnClangFormat(args, done, stdio) {
               "Consider installing it with your native package manager instead.\n";
     throw new Error(message);
   }
-  var clangFormatProcess = spawn(nativeBinary, args, {stdio: stdio});
-  clangFormatProcess.on('close', function(exit) {
-    if (exit) done(exit);
-  });
-  return clangFormatProcess;
+
+  // extract glob, if present
+  var filesGlob = args.filter(function(arg){return arg.indexOf('--glob=') !== -1;})
+                    .map(function(arg){return arg.replace('--glob=', '');})
+                    .shift();
+
+  if (filesGlob) {
+    // remove glob from arg list
+    args = args.filter(function(arg){return arg.indexOf('--glob=') === -1;});
+
+    return glob(filesGlob, function(er, files) {
+      // launch a new process for each file, up to 200 at a time
+      async.parallelLimit(
+        files.map(function(file) {
+          return function(callback) {
+            var clangFormatProcess = spawn(nativeBinary,
```

Comment:
The correct fix is to write the paths into a params file, and hand
clang-format a reference to it. But that doesn't seem to be supported.
Your reasoning is good, 30 SGTM

On Fri, Feb 12, 2016 at 6:53 AM Filipe Silva <notifications@github.com>
wrote:

> In index.js
> <https://github.com/angular/clang-format/pull/23#discussion_r52749905>:
>
> > +
> > +  // extract glob, if present
> > +  var filesGlob = args.filter(function(arg){return arg.indexOf('--glob=') !== -1;})
> > +                    .map(function(arg){return arg.replace('--glob=', '');})
> > +                    .shift();
> > +
> > +  if (filesGlob) {
> > +    // remove glob from arg list
> > +    args = args.filter(function(arg){return arg.indexOf('--glob=') === -1;});
> > +
> > +    return glob(filesGlob, function(er, files) {
> > +      // launch a new process for each file, up to 200 at a time
> > +      async.parallelLimit(
> > +        files.map(function(file) {
> > +          return function(callback) {
> > +            var clangFormatProcess = spawn(nativeBinary,
>
> Windows seems to have a 260 char path length max, and 8191 char cli
> command max. Dividing one by the other gives 30, so maybe that's reasonable
> safe value?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/clang-format/pull/23/files#r52749905>.
>